db.go

```diff
@@ -383,11 +383,15 @@ func (db *DB) close() error {
 	if !db.opened {
 		return nil
 	}
-	
+
 	db.opened = false
 
 	db.freelist = nil
-	db.path = ""
+
+	// defer clearing the path because funlock may need it.
+	defer func() {
+		db.path = ""
+	}()
```

Comment:
There's not an explicit guarantee about `path` when an error occurs. Honestly, I wish I hadn't reset the `path` at all on `Close()` but that's behavior now. I'm OK with changing the behavior on error to not clear the path.